app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactsFragmentTest.kt

```diff
@@ -16,23 +16,50 @@ import androidx.test.uiautomator.UiSelector
 import com.github.factotum_sdp.factotum.MainActivity
 import com.github.factotum_sdp.factotum.R
 import com.github.factotum_sdp.factotum.placeholder.ContactsList
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
 import junit.framework.TestCase.assertTrue
+import kotlinx.coroutines.runBlocking
 import org.junit.Before
+import org.junit.BeforeClass
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class ContactsFragmentTest {
+    private val WAITFORANIMATION = 500L
+
     @get:Rule
-    var activityRule = ActivityScenarioRule(MainActivity::class.java)
+    var testRule = ActivityScenarioRule(
+        MainActivity::class.java
+    )
+
+    companion object {
+        @BeforeClass
+        @JvmStatic
+        fun setUpDatabase() {
+            val database = Firebase.database
+            database.useEmulator("10.0.2.2", 9000)
+            MainActivity.setDatabase(database)
+
+            emptyFirebaseDatabase(database)
+
+            ContactsList.init(database)
 
+            runBlocking {
+                ContactsList.populateDatabase()
+            }
+        }
+    }
     @Before
-    fun goToContacts() {
+    fun setUp() {
```

Comment:
Code could use some formatting.
Try Ctrl + Alt + L.

Reply:
I agree with you on that, the command was run 👍